src/engine/strat_engine/filesystem.rs

```diff
@@ -64,6 +97,33 @@ impl StratFilesystem {
         self.thin_dev.id()
     }
 
+    /// Return an target size for the filesystem over the thin device.
+    /// TODO: returning 2 times the current size will double the space provisoned to
+    /// the filesystem.  We should determine if this is a reasonable value.
+    fn extend_size(&self, current_size: DataBlocks) -> DataBlocks {
```

Comment:
You already have a function called ```extend_size``` in StratPool which returns the amount by which to extend the size. If this function does something different, it should have a different name.